lib/matplotlib/axes/_axes.py

```diff
@@ -6407,34 +6407,11 @@ def _normalize_input(inp, ename='input'):
 
             # adopted from adjust_x/ylim part of the bar method
             if orientation == 'horizontal':
-                xmin0 = max(_saved_bounds[0]*0.9, minimum)
                 xmax = self.dataLim.intervalx[1]
-                for m in n:
-                    # make sure there are counts
-                    if np.sum(m) > 0:
-                        # filter out the 0 height bins
-                        xmin = np.amin(m[m != 0])
-                    # If no counts, set min to zero
-                    else:
-                        xmin = 0.0
-                xmin = max(xmin*0.9, minimum) if not input_empty else minimum
-                xmin = min(xmin0, xmin)
-                self.dataLim.intervalx = (xmin, xmax)
+                self.dataLim.intervalx = (0, xmax)
```

Comment:
Yes, everything can be made to work, at least under reasonable circumstances... but the devil is in the details.  I am making progress but running into problems with the test framework.  Rebasing on current master helps, I think.  Do you mind if I force-push the rebase?